betty/plugin/__init__.py

```diff
@@ -84,16 +78,16 @@ def cls(self) -> builtins.type[_PluginCoT]:
         assert self._cls is not None
         return self._cls
 
-    def __call__(self, cls: builtins.type[_PluginCoT]) -> builtins.type[_PluginCoT]:
+    def __call__(self, cls: builtins.type[_T]) -> builtins.type[_T]:
```

Comment:
Extract the decoration functionality of both definition classes into a new private superclass.